pkg/kv/kvserver/kvstorage/init.go

```diff
@@ -416,19 +415,6 @@ func loadReplicas(ctx context.Context, eng storage.Engine) ([]Replica, error) {
 
 	// INVARIANT: all replicas have a persisted full ReplicaID (i.e. a "ReplicaID from disk").
 	//
-	// This invariant is true for replicas created in 22.1. Without further action, it
-	// would be violated for clusters that originated before 22.1. In this method, we
-	// backfill the ReplicaID (for initialized replicas) and we remove uninitialized
-	// replicas lacking a ReplicaID (see below for rationale).
-	//
-	// The migration can be removed when the KV host cluster MinSupportedVersion
-	// matches or exceeds 23.1 (i.e. once we know that a store has definitely
-	// started up on >=23.1 at least once).
-
-	// Collect all the RangeIDs that either have a RaftReplicaID or HardState. For
-	// unmigrated replicas we see only the HardState - that is how we detect
-	// replicas that still need to be migrated.
-	//
```

Comment:
This comment is removed but not the block of code that it comments. Leave a comment what this block does?